tests/ops/functions/test_iterative_qpe.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the iterative_qpe function
+"""
+import pytest
+
+import numpy as np
+import pennylane as qml
+
+
+class TestResult:
+    """Test to check that the results obtained with iterative_qpe are equivalent to those of QuantumPhaseEstimation"""
+
+    @pytest.mark.parametrize("phi", (1.0, 2.0, 3.0))
+    def test_compare_qpe(self, phi):
+        """Test to check that the results obtained are equivalent to those of QuantumPhaseEstimation"""
+
+        # TODO: When we have general statistics on measurements we can calculate it exactly with qml.probs
+        dev = qml.device("default.qubit", shots=10000000)
+
+        @qml.qnode(dev)
+        def circuit_iterative():
+            # Initial state
+            qml.PauliX(wires=[0])
+
+            # Iterative QPE
+            measurements = qml.iterative_qpe(qml.RZ(phi, wires=[0]), estimation_wire=[1], iters=3)
+
+            return [qml.sample(op=meas) for meas in measurements]
+
+        sample_list = np.array(circuit_iterative())
+        sample_list = sample_list.T
+        output = qml.probs().process_samples(np.array(sample_list), wire_order=[0, 1, 2])
+
+        @qml.qnode(dev)
+        def circuit_qpe():
+            # Initial state
+            qml.PauliX(wires=[0])
+
+            # Iterative QPE
+            qml.QuantumPhaseEstimation(qml.RZ(phi, wires=[0]), estimation_wires=[1, 2, 3])
+
+            return qml.probs(wires=[1, 2, 3])
+
+        assert np.allclose(np.round(output, 2), np.round(circuit_qpe(), 2))
+
+    @pytest.mark.jax
+    def test_check_gradients(self):
```

Comment:
Is there anyway to compare this with QPE to ensure the computed gradient makes sense?

Reply:
I have not been able to get equal results. Being equivalent under certain conditions, I have the feeling that the gradients should not coincide. I think it is an interesting topic some researcher could study with PennyLane now :)